client/src/pages/Admin/Admin.tsx

```diff
@@ -1,90 +1,86 @@
 import { useState } from 'react';
+import { useUsersQuery } from '../../generated/graphql-types';
 import UserList from '../../components/UserList/UserList';
 import SearchBar from '../../components/SearchBar/SearchBar';
 import OptionSelect from '../../components/OptionSelect/OptionSelect';
-import users from './fakeUsers';
 
 export default function Admin() {
+  const { data, loading, error } = useUsersQuery();
   const [searchByName, setSearchByName] = useState<string>('');
   const [role, setRole] = useState<string>('');
 
   const handleChange = (value: string): void => {
     setSearchByName(value);
   };
 
-  // function to filter by name, first name depending on the role
-  const filteredUsers = users.filter(
-    (user) =>
-      user.role.toLowerCase().includes(role.toLowerCase()) &&
-      (user.name.toLowerCase().includes(searchByName.toLowerCase()) ||
-        user.lastName.toLowerCase().includes(searchByName.toLowerCase()))
-  );
+  const filteredUsers =
+    data?.users?.filter(
+      (user) =>
+        user.role.label.toLowerCase().includes(role.toLowerCase()) &&
+        (user.firstname.toLowerCase().includes(searchByName.toLowerCase()) ||
+          user.lastname.toLowerCase().includes(searchByName.toLowerCase()))
+    ) || [];
+
+  if (loading) return <h1>Chargement ...</h1>;
+  if (error) return <h1>Erreur</h1>;
```

Comment:
I think we should be more explicit on this error, if we choose to display it to users, we should for example say they can try to refresh the page and if the error persists [...]

Reply:
C'est oasis bête effectivement juste erreur c'est peut être un peut lite 
Bien vu